caas/kubernetes/provider/modeloperator.go

```diff
@@ -549,6 +528,12 @@ func ensureModelOperatorRBAC(
 	labels map[string]string,
 ) (string, []func(), error) {
 	cleanUpFuncs := []func(){}
+	ctx := context.TODO()
+
+	k8sClient := broker.Client()
+	if k8sClient == nil {
+		return operatorName, cleanUpFuncs, errors.New("kubernetes client cannot be nil")
+	}
```

Comment:
this check should be done when the broker is constructed
`broker.Client()` should never return nil